oracle/datadog_checks/oracle/oracle.py

```diff
@@ -156,9 +157,23 @@ def _connection(self):
                     self.log.error("The JDBC connection failed with the following error: %s", str(e))
                     self._connection_errors += 1
             else:
+                if self._use_instant_client:
+                    self.init_instant_client()
+                else:
+                    self.log.debug('Connecting to Oracle using the native client')
                 self._cached_connection = self._oracle_connect()
         return self._cached_connection
 
+    def init_instant_client(self):
+        try:
+            oracledb.init_oracle_client()
```

Comment:
what happens if this is called by multiple instances?

Reply:
From documentation: The init_oracle_client() method can be called multiple times in each Python process as long as the arguments are the same each time.
https://python-oracledb.readthedocs.io/en/latest/api_manual/module.html#oracledb.init_oracle_client